### ISSUE TITLE
[Bug]: Unable to join worlds after deleting the config files, possible incompat with Enchancement

### ISSUE THREAD

**Issue by SplendidAlakey** [Fuzss/universalenchants#36]
### Mod Loader (Required)

Fabric

### Minecraft Version (Required)

1.19.2

### Mod Version (Required)

4.2.9

### Notes (Required)

1. Create a world, let the configs generate
2. Close the game, delete .mineacraft/config/universalenchants folder entirely
3. Install [Enchancement](https://www.curseforge.com/minecraft/mc-mods/enchancement) (not sure if any other mod would cause this, I tested with [Qu Enchantments](https://github.com/Quplet/Qu-Enchantments) and it worked fine)
4. Launch the game and try to join the world you created in step 1

The game window will tell you, that errors in currently selected datapacks prevented the world from loading. However, from the logs you can clearly see, that UE simply failed to regenerate the config files.

### latest.log (Optional)

https://gist.github.com/SplendidAlakey/42f448cd46061668d74102c691d4de84

**Comment by Fuzss** [comment]
Can you share your config/s for Enchancement?

**Comment by SplendidAlakey** [comment]
Reproducible with default configs. The logs I shared are from an instance, that was freshly created specifically for the test.

<details><summary>The default config looks like this:</summary>

```
{
  "allowedEnchantments": [
    "enchancement:assimilation",
    "enchancement:perception",
    "enchancement:veil",
    "enchancement:amphibious",
    "enchancement:strafe",
    "enchancement:wardenspine",
    "enchancement:dash",
    "enchancement:impact",
    "enchancement:slide",
    "enchancement:acceleration",
    "enchancement:bouncy",
    "enchancement:gale",
    "enchancement:berserk",
    "minecraft:fire_aspect",
    "enchancement:frostbite",
    "minecraft:infinity",
    "enchancement:chaos",
    "enchancement:delay",
    "enchancement:phasing",
    "enchancement:brimstone",
    "enchancement:homing",
    "enchancement:torch",
    "minecraft:channeling",
    "enchancement:leech",
    "minecraft:riptide",
    "enchancement:warp",
    "enchancement:extracting",
    "enchancement:molten",
    "enchancement:beheading",
    "enchancement:lumberjack",
    "enchancement:bury",
    "enchancement:scooping",
    "enchancement:disarm",
    "enchancement:grapple",
    "minecraft:luck_of_the_sea",
    "minecraft:efficiency",
    "minecraft:silk_touch"
  ],
  "invertedList": false,
  "overhaulEnchantingTable": true,
  "allowTreasureEnchantmentsInEnchantingTable": false,
  "experienceLevelCost": 5,
  "lapisLazuliCost": 5,
  "singleEnchantmentMode": true,
  "singleLevelMode": true,
  "allowInfinityOnCrossbows": true,
  "allTridentsHaveLoyalty": true,
  "channelingWorksWhenNotThundering": true,
  "fireAspectWorksAsFlintAndSteel": true,
  "freeEnchantedBookMerging": true,
  "luckOfTheSeaHasLure": true,
  "negateEnderPearlDamage": true,
  "safeChanneling": true,
  "tridentsReturnFromVoid": true,
  "weakerPotions": true,
  "weaponEnchantmentCooldownRequirement": 0.7,
  "maxExtractingBlocks": 64,
  "maxLumberjackBlocks": 1024,
  "maxLumberjackHorizontalLength": 7,
  "unbreakingChangesFlag": 0
}
```

</details>

**Comment by Fuzss** [comment]
Ok, so I looked into this and it turns out Enchancement removes (more like hides) enchantments it wants to disable from the game registry. This is a very incompatible approach for this sort of behavior, I'm not going to support that in any way.

You can configure Enchancement to keep all vanilla enchantments in the game though (check their FAQ on CurseForge), in that case Universal Enchants will work again, I just tested it myself.

**Comment by Fuzss** [comment]
I made a minor change in v4.2.10, Enchancement should work out of the box now.

**Comment by SplendidAlakey** [comment]
Unfortunately, it still doesn't work. New logs (same link): https://gist.github.com/SplendidAlakey/42f448cd46061668d74102c691d4de84

**Comment by Fuzss** [comment]
Fixed for real now, just tested it myself in v4.2.11.

**Comment by SplendidAlakey** [comment]
Yep, now it works and it's much better organized, thank you!
